File: src/skills/stack.java
package skills;

import java.util.Stack;

public class stack {

    public static void main(String[] args) {
        Stack<Integer> st = new Stack<>();
        System.out.println(st.empty());
        st.push(1);
        st.push(2);
        st.push(3);
        int size = st.size();
        System.out.println("size = " + size);
        Integer last = st.pop();
        System.out.println("last = " + last);
        System.out.println("size = " + st.size());
        System.out.println(st.isEmpty());
    }
}

File: boki-algorithm/src/inflearn/dingco/section3/stack/stack.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None


class Stack:
    def __init__(self):
        self.head = None

    def push(self, value):
        new_head = Node(value)
        new_head.next = self.head
        self.head = new_head

    def pop(self):
        if self.is_empty():
            return "stack is Empty"
        deleted_head = self.head
        self.head = self.head.next
        return deleted_head

    def peek(self):
        if self.is_empty():
            return "stack is Empty"
        return self.head.data

    def is_empty(self):
        return self.head is None

stack = Stack()
stack.push(4)
print(stack.peek()) # stack.head.data

stack.push(3)
print(stack.peek())

stack.push(5)
print(stack.peek())

print()

stack.pop()
print(stack.peek())

stack.pop()
print(stack.peek())

stack.pop()
print(stack.peek())
